src/query/config/src/config.rs

```diff
@@ -1859,3 +1810,405 @@ impl TryInto<InnerLocalConfig> for LocalConfig {
+                path: value.path,
+            })
+        }
+    }
+
+    impl From<inner::DiskCacheConfig> for DiskCacheConfig {
+        fn from(value: inner::DiskCacheConfig) -> Self {
+            Self {
+                max_bytes: value.max_bytes,
+                path: value.path,
+            }
+        }
+    }
+
+    impl TryFrom<CacheStorageTypeConfig> for inner::CacheStorageTypeConfig {
+        type Error = ErrorCode;
+        fn try_from(value: CacheStorageTypeConfig) -> std::result::Result<Self, Self::Error> {
+            Ok(match value {
+                CacheStorageTypeConfig::None => inner::CacheStorageTypeConfig::None,
+                CacheStorageTypeConfig::Disk => inner::CacheStorageTypeConfig::Disk,
+            })
+        }
+    }
+
+    impl From<inner::CacheStorageTypeConfig> for CacheStorageTypeConfig {
+        fn from(value: inner::CacheStorageTypeConfig) -> Self {
+            match value {
+                inner::CacheStorageTypeConfig::None => CacheStorageTypeConfig::None,
+                inner::CacheStorageTypeConfig::Disk => CacheStorageTypeConfig::Disk,
+            }
+        }
+    }
+}
+
+// Obsoleted configuration entries checking
+//
+// The following code should be removed from the release after the next release.
+// Just give user errors without any detail explanation and migration suggestions.
+impl QueryConfig {
+    fn check<T>(
+        target: &Option<T>,
+        cli_arg_name: &str,
+        alternative_cli_arg: &str,
+        alternative_toml_config: &str,
+        alternative_env_var: &str,
+    ) -> Option<String> {
+        target.as_ref().map(|_| {
+            format!(
+                "
+ --------------------------------------------------------------
+ *** {cli_arg_name} *** is obsoleted : 
+ --------------------------------------------------------------
+   alternative command-line options : {alternative_cli_arg}
+   alternative environment variable : {alternative_env_var}
+            alternative toml config : {alternative_toml_config}
+ --------------------------------------------------------------
+"
+            )
+        })
+    }
+    fn check_obsoleted(&self) -> Result<()> {
+        let check_results = vec![
+            Self::check(
+                &self.table_disk_cache_mb_size,
+                "table-disk-cache-mb-size",
+                "cache-disk-max-bytes",
+                r#"
+                    [cache]
+                    ...
+                    data_cache_storage = "disk"
+                    ...
+                    [cache.disk]
+                    max_bytes = [MAX_BYTES]  
+                    ...
+                  "#,
+                "CACHE_DISK_MAX_BYTES",
+            ),
+            Self::check(
+                &self.table_meta_cache_enabled,
+                "table-meta-cache-enabled",
+                "cache-enable-table-meta-caches",
+                r#"
+                    [cache]
+                    table-meta-cache-enabled=[true|false]
```

Comment:
`table-meta-cache-enabled=[true|false]` -> `enable_table_meta_caches=[true|false]`

Reply:
Fixed in https://github.com/dantengsky/fuse-query/pull/79